modules/account-lib/src/coin/algo/assetTransferBuilder.ts

```diff
@@ -1,26 +1,171 @@
+   */
+  assetIndex(index: number): this {
+    if (index <= 0) {
+      throw new Error('Asset index must be a uint64 value');
+    }
+    this._assetIndex = index;
+
+    return this;
+  }
+
+  /**
+   * Sets the asset amount to be transferred.
+   *
+   * @param {BigNumber} amount The amount to be transferred.
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://developer.algorand.org/docs/reference/transactions/#asset-transfer-transaction
+   */
+  assetAmount(amount: BigNumber): this {
+    if (amount.lt(0)) {
+      throw new Error('transfer amount cannot be less than 0');
+    }
+    this._assetAmount = BigInt(amount.toString());
+
+    return this;
+  }
+
+  /**
+   * Sets the receiver of the asset transfer.
+   *
+   * @param {BaseAddress} receiver The receiver of the asset transfer.
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://developer.algorand.org/docs/reference/transactions/#asset-transfer-transaction
+   */
+  receiver(receiver: BaseAddress): this {
+    this.validateAddress(receiver);
+    this._receiver = receiver.address;
+
+    return this;
+  }
+
+  /**
+   * Sets the closeTo address.
+   *
+   * If set the any remaining assets in the sender's account will be closed to
+   * this address.
+   *
+   * @param {BaseAddress} closeTo The closeTo address.
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://developer.algorand.org/docs/reference/transactions/#asset-transfer-transaction
+   */
+  closeTo(closeTo: BaseAddress): this {
+    this.validateAddress(closeTo);
+    this._closeTo = closeTo.address;
+
+    return this;
+  }
+
+  /**
+   * Sets the parameters of the transaction builder to allowlist an asset.
+   *
+   * To allow list an asset, you send 0 units of the asset to yourself.
+   *
+   * This method sets the assetIndex, sender, receiver, asset amount, and
+   * fee parameters to their respective values to allowlist and asset.
+   *
+   * @param {number} assetIndex The unique identifier of the asset.
+   * @param {BaseAddress} userAddress The address of the user.
+   * @returns {TransactionBuilder} This transaction builder.
+   */
+  allowListAsset(assetIndex: number, userAddress: BaseAddress): this {
+    this.assetIndex(assetIndex);
+    this.sender(userAddress);
+    this.receiver(userAddress);
+    this.isFlatFee(true);
+    this.fee({ fee: '1000' });
+    this._assetAmount = 0;
+
+    return this;
+  }
+
   /** @inheritdoc */
-  protected buildImplementation(): Promise<Transaction> {
-    throw new NotImplementedError('buildImplementation not implemented');
+  protected async buildImplementation(): Promise<Transaction> {
+    this._transaction.setAlgoTransaction(
+      algosdk.makeAssetTransferTxnWithSuggestedParams(
+        this._sender,
+        this._receiver,
+        this._closeTo,
+        undefined,
+        this._assetAmount,
+        this._note,
+        this._assetIndex,
+        {
```

Comment:
can we suggested params from TransactionBuilder?

Reply:
ah yeah forgot about that